server.minimal.Dockerfile

```diff
@@ -1,19 +1,38 @@
+FROM golang:1.9.4-alpine AS build-env
+RUN apk add --update git gcc libc-dev
```

Comment:
If "tiny-ness" is important; use `--no-cache` instead;

```bash
docker build -t before -<<EOF
FROM golang:1.9.4-alpine AS build-env
RUN apk add --update git gcc libc-dev
EOF

docker build -t after -<<EOF
FROM golang:1.9.4-alpine AS build-env
RUN apk add --no-cache git gcc libc-dev
EOF
```

```
REPOSITORY      CREATED                  SIZE
after           Less than a second ago   386MB
before          Less than a second ago   387MB
```

Reply:
This is using a build image and a final image, though - does the no-cache affect the final image?  This is the before and after image built with the `server.minimal.Dockerfile`, without the `--no-cache` option and with the `--no-cache` option:

```
REPOSITORY                                       TAG                    IMAGE ID            CREATED              SIZE
<none>                                           <none>                 fa5c316b8562        17 seconds ago       540MB
after                                            latest                 f3f0621261c3        About a minute ago   21.1MB
before                                           latest                 f3f0621261c3        About a minute ago   21.1MB
<none>                                           <none>                 82a30458273b        About a minute ago   540MB
```

Reply:
No it doesn't affect the final image (but does affect your local cache 😅); it was mainly a nit, and I wanted to mention `--no-cache` because it's awesome: it takes care of everything (updating the index, and removing it afterwards)

Reply:
Oh nice!  It seems to only be a MB or less, so in this case it seems ok, but I will keep that in mind for all my other images!